nodebuilder/p2p/metrics.go

```diff
@@ -0,0 +1,171 @@
+package p2p
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/metrics"
+	"github.com/libp2p/go-libp2p/core/network"
+	rcmgr "github.com/libp2p/go-libp2p/p2p/host/resource-manager"
+	rcmgrObs "github.com/libp2p/go-libp2p/p2p/host/resource-manager/obs"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+)
+
+// global meter provider (see opentelemetry docs)
+var (
+	meter = global.MeterProvider().Meter("p2p")
+)
+
+// WithInfoMetrics option sets up metrics for p2p networking.
+func WithInfoMetrics(bc *metrics.BandwidthCounter) error {
+	bandwidthTotalInbound, err := meter.
+		SyncInt64().
+		Histogram(
+			"p2p_bandwidth_total_inbound",
+			instrument.WithUnit(unit.Bytes),
+			instrument.WithDescription("total number of bytes received by the host"),
+		)
+	if err != nil {
+		return err
+	}
+	bandwidthTotalOutbound, _ := meter.
+		SyncInt64().
+		Histogram(
+			"p2p_bandwidth_total_outbound",
+			instrument.WithUnit(unit.Bytes),
+			instrument.WithDescription("total number of bytes sent by the host"),
+		)
+	if err != nil {
+		return err
+	}
+	bandwidthRateInbound, _ := meter.
+		SyncFloat64().
+		Histogram(
+			"p2p_bandwidth_rate_inbound",
+			instrument.WithDescription("total number of bytes sent by the host"),
+		)
+	if err != nil {
+		return err
+	}
+	bandwidthRateOutbound, _ := meter.
+		SyncFloat64().
+		Histogram(
+			"p2p_bandwidth_rate_outbound",
+			instrument.WithDescription("total number of bytes sent by the host"),
+		)
+	if err != nil {
+		return err
+	}
+	p2pPeerCount, _ := meter.
+		AsyncFloat64().
+		Gauge(
+			"p2p_peer_count",
+			instrument.WithDescription("number of peers connected to the host"),
+		)
+	if err != nil {
+		return err
+	}
+
+	if err = meter.RegisterCallback(
+		[]instrument.Asynchronous{
+			p2pPeerCount,
+		}, func(ctx context.Context) {
+			bcStats := bc.GetBandwidthTotals()
```

Comment:
You can get stats per protocol by calling `GetBandwidthByProtocol`. That could allow you add extra `protocol` attribute to stats